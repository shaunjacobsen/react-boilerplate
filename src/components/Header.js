import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from './../actions/auth';

export const Header = props => (
  <header className="header">
    <div className="content-container">
    <div className="header__content">
      <Link to="/dashboard" className="header__title">
        <h1>Boilerplate</h1>
      </Link>
      <button onClick={props.startLogout} className="button button--link">Log out</button>
      </div>
    </div>
  </header>
);

const mapDispatchToProps = dispatch => {
  return {
    startLogout: () => dispatch(startLogout()),
  };
};

export default connect(undefined, mapDispatchToProps)(Header);
