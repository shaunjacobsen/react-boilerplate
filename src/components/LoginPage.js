import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from './../actions/auth';

export const LoginPage = props => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Boilerplate</h1>
      <p>Tagline</p>
      <button className="button" onClick={props.startLogin}>Login with Google</button>
    </div>
  </div>
);

const mapDispatchToProps = dispatch => {
  return {
    startLogin: () => dispatch(startLogin()),
  };
};

export default connect(undefined, mapDispatchToProps)(LoginPage);
