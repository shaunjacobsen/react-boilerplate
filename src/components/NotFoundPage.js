import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
  <div>
    <h1>404</h1>
    <p><Link to='/'>Home</Link></p>
  </div>
);

export default NotFoundPage;