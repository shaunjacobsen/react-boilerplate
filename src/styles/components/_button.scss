.button {
  background-color: $blue;
  border: none;
  border-radius: 3px;
  color: #fff;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button--grey {
  background-color: grey;
}

.button--link {
  background-color: none;
}