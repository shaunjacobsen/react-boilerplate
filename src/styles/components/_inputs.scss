.text-input {
  border: 1px solid #cacccd;
  font-size: $font-size-large;
  font-weight: 300;
  height: 50px;
  padding: $s-size;
}

.select {
  @extend .text-input;
  border-radius: 0;
  -webkit-appearance: none;
  -webkit-border-radius: 0px;
}

.text-area {
  @extend .text-input;
  height: 10rem;
}