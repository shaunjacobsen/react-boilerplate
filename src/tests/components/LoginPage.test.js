import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from './../../components/LoginPage';

test('it should render the LoginPage component properly', () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});

test('it should call startLogin on log in button click', () => {
  const startLogin = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={startLogin} />);
  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalled();
});