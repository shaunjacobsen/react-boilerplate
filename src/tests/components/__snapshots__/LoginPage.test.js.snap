// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render the LoginPage component properly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Boilerplate
    </h1>
    <p>
      Tagline
    </p>
    <button
      className="button"
    >
      Login with Google
    </button>
  </div>
</div>
`;
